Ease.DryIoc/DryIocContainerTestBase.cs

```diff
@@ -1,46 +1,50 @@
-﻿using System;
-using DryIoc;
+﻿using DryIoc;
 using Moq;
+using System;
 
-namespace Ease.NUnit.DryIoc
+namespace Ease.DryIoc
 {
-	public class DryIocContainerTestBase : ContainerTestBase
+    public abstract class DryIocContainerTestBase : ContainerTestBase
 	{
-		private Container Container;
-		private IResolverContext ScopeContext;
+		protected Container _container;
+		protected IResolverContext _scopeContext;
```

Comment:
Also, no `_` please. camelCase for private fields, PascalCase for anything public (including protected) or private properties. will update with more later, will add editor config and resharper configs at some point. (yes, I know I wasn't following these rules, but I put this together pretty quick)

Reply:
I see where this is being used in the unit test framework specific classes for the `RegisterPerTestSetup`. What I'd like is to have a protected method here that is called from those classes
```
protected void ResetLifetime()
{
  ScopeContext?.Dispose();
  ScopeContext = Container.OpenScope();
}
```

Reply:
> still going through this, was there a reason to make this protected? I'm trying to keep any test code from direct access to the container (everything should use one of the `Register(Mock)Type` methods or `ResolveType` methods .

Understand and agree, will have a think. I made it protected for a couple of reasons:
- `NUnit` needed access to the scope context for per test setup
- `MSTest` has very different setup requirements to both `NUnit` and `xUnit`

Reply:
Had a quick look through the `MSTest` documentation and found that it's possible to have a similar setup to `NUnit` by using the `TestInitialize` attribute. Have updated to use this - if this isn't to your liking it can easily be reverted.